shell/platform/linux/fl_texture_registrar.cc

```diff
@@ -57,21 +60,27 @@ static void engine_weak_notify_cb(gpointer user_data,
   self->engine = nullptr;
 
   // Unregister any textures.
+  g_mutex_lock(&self->textures_mutex);
   g_autoptr(GHashTable) textures = self->textures;
   self->textures = g_hash_table_new_full(g_direct_hash, g_direct_equal, nullptr,
                                          g_object_unref);
   g_hash_table_remove_all(textures);
+  g_mutex_unlock(&self->textures_mutex);
+  g_mutex_clear(&self->textures_mutex);
```

Comment:
The mutex shouldn't be cleared here - it will be done in dispose. It may still be accessed before that.